code/game/machinery/firealarm.dm

```diff
@@ -326,3 +326,121 @@
+
+/obj/machinery/sprinkler/interact(mob/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='danger'>Access denied.</span>")
+		return
+	dispense()
+
+/obj/machinery/sprinkler/emag_act(mob/user)
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return
+	playsound(src, "sparks", 75, 1)
+	obj_flags |= EMAGGED
+	internal.reagents.remove_all(50)
+	internal.reagents.add(pick(/datum/reagent/clf3,
+							   /datum/reagent/drug/space_drugs,
+							   /datum/reagent/lube,
+							   /datum/reagent/fuel/unholywater,
+							   /datum/reagent/toxin/acid/fluacid,
+							   /datum/reagent/napalm),50)
+	to_chat(user, "<span class='warning'>The war crime LED blinks twice.</span>")
+
+/obj/machinery/sprinkler/fire_act()
+	.=..()
+	dispense()
+
+/obj/machinery/sprinkler/proc/dispense()
+	if(cooldown < world.time && refill_loaded)
+		var/datum/effect_system/smoke_spread/chem/smoke_machine/smoke = new()
+		smoke.set_up(internal.reagents, 5, 8,  get_turf(src))
+		smoke.start()
+		cooldown = world.time + 200
+		if(!internal.reagents)
+			refill_loaded = FALSE
+			update_icon()
+
+/obj/machinery/sprinkler/update_icon()
+	. = ..()
+	if(refill_loaded)
+		icon_state = "sprinkler1"
+		return
+	icon_state = "sprinkler0"
+
+/obj/item/sprinkler_refill
+	name = "sprinkler refill (water)"
+	desc = "Apply this refill if the sprinkler light is red."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "sprinkler_refill_water"
+	w_class = WEIGHT_CLASS_SMALL
+	var/obj/item/reagent_containers/glass/beaker/refill
+	var/used = FALSE
+	var/chemtype = /datum/reagent/water
+
+/obj/item/sprinkler_refill/Initialize()
+	. = ..()
+	refill = new /obj/item/reagent_containers/glass/beaker(src)
+	refill.reagents.add_reagent(chemtype, 50)
+
+/obj/item/sprinkler_refill/examine(mob/user)
+	. = ..()
+	if(used)
+		. += "It has been used up, better throw it in the trash."
+
+/obj/item/sprinkler_refill/update_icon()
+	. = ..()
+	cut_overlays()
+	if(used)
+		add_overlay("refill_empty")
+		return
+	add_overlay("refill_[max(round(refill.reagents.total_volume/10),1)]")
+
+/obj/item/sprinkler_refill/attack_obj(obj/O, mob/living/user)
+	if(!used && istype(O, /obj/machinery/sprinkler))
+		var/obj/machinery/sprinkler/S = O
+		refill.reagents.trans_to(S.internal, 50, transfered_by = user)
+		playsound(loc, 'sound/effects/refill.ogg', 50, 1, -6)
+		to_chat(user, "<span class='warning'>You refill the [S]</span>")
+		S.refill_loaded = TRUE
+		S.update_icon()
+		if(!refill.reagents.total_volume)
+			used = TRUE
+		update_icon()
+
+/obj/item/sprinkler_refill/foam
+	name = "sprinkler refill (firefighting foam)"
+	icon_state = "sprinkler_refill_foam"
+	chemtype = /datum/reagent/firefighting_foam
+
+/obj/item/deployable_sprinkler
+	name = "deployable sprinkler"
+	desc = "A self-deploying sprinkler, just press the button to activate it."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "sprinkler_d"
+
+/obj/item/deployable_sprinkler/attack_self(mob/user)
+	. = ..()
+	to_chat(user, "You start the deployment the [src]...")

```

Comment:
You start the deployment of \the [src]

Also the one two lines below should be \the [src] as well